### PR TITLE
Add FeatureMetadata related APIs

### PR DESCRIPTION
**Pull request by imingtsou**
*Issue #, if available:*
Add new launched APIs -- UpdateFeatureMetadata and DescribeFeatureMetadata.

*Description of changes:*
Add new launched APIs -- UpdateFeatureMetadata and DescribeFeatureMetadata.

*Testing done:*
Tested with SageMaker Studio.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [X] I have tested my notebook(s) and ensured it runs end-to-end
- [X] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jkroll-aws — APPROVED** [review]

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 1]
```diff
@@ -24,21 +24,21 @@
     "This notebook uses both `boto3` and Python SDK libraries, and the `Python 3 (Data Science)` kernel. This notebook works with Studio, Jupyter, and JupyterLab. \n",
     "\n",
     "#### Library dependencies:\n",
-    "* sagemaker>=2.0.0\n",
-    "* numpy\n",
+    "* sagemaker>=2.100.0\n",
+    "* NumPy\n",
```
Comment (alex-tang) [inline sagemaker-featurestore/feature_store_introduction.ipynb:28]: I think the dependency names are based off of the python package (i.e. ones listed on PyPI). It should be kept as numpy.

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 2]
```diff
@@ -318,6 +310,83 @@
     "check_feature_group_status(orders_feature_group)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Add metadata to a feature\n",
+    "\n",
+    "We can put some searchable metadata to the features of the FeatureGroup by using the `UpdateFeatureMetadata` API. The current support metadata fields are `description` and `parameters`."
```
Comment (alex-tang) [inline sagemaker-featurestore/feature_store_introduction.ipynb:319]: The _currently supported_ metadata fields.

Comment (alex-tang) [inline sagemaker-featurestore/feature_store_introduction.ipynb:319]: We can _add searchable metadata fields to FeatureGroup features_

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 3]
```diff
@@ -318,6 +310,83 @@
...
+    "    description=\"The ID of a customer. It is also used in orders_feature_group.\",\n",
+    "    parameter_additions=[FeatureParameter(\"idType\", \"primaryKey\")],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To confirm that the feature has been updated with the new metadata, we use `DescribeFeatureMetadata` to display that feature."
```
Comment (alex-tang) [inline sagemaker-featurestore/feature_store_introduction.ipynb:341]: I wouldn't say 'to confirm .. has been updated'. The update_feature_metadata operation is synchronous and the user of this notebook will know if the operation succeeds or fails. 

Instead, consider phrasing it as 'To view feature metadata, we can use ...'

File: sagemaker-featurestore/feature_store_introduction.ipynb [hunk 4]
```diff
@@ -318,6 +310,83 @@
...
+   "outputs": [],
+   "source": [
+    "customers_feature_group.describe_feature_metadata(feature_name=\"customer_id\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Feature metadata fields are searchable. We use `search` API with filters to display the specific feature."
```
Comment (alex-tang) [inline sagemaker-featurestore/feature_store_introduction.ipynb:357]: I would rephrase the second sentence to _describe the use case_ this feature is meant to support more accurately.

'with filters to display the specific feature' should be replaced with 'find features with metadata that matches some search criteria'.

File: sagemaker-featurestore/feature_store_introduction.ipynb (hunk 1 above)
Comment (imingtsou) [reply sagemaker-featurestore/feature_store_introduction.ipynb:28]: Changed this because numpy will fail grammar check. Better to use code format here. Update in the next PR.
